packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -369,12 +393,21 @@ export class RecordSet extends Resource implements IRecordSet {
 
     if (this.weight) {
       const idPrefix = `WEIGHT_${this.weight}_ID_`;
-      const identifier = `${idPrefix}${Names.uniqueResourceName(this, { maxLength: 64 - idPrefix.length })}`;
-      return identifier;
+      return this.createIdentifier(idPrefix);
+    }
+
+    if (this.region) {
+      const idPrefix= `REGION_${this.region}_ID_`;
+      return this.createIdentifier(idPrefix);
     }
 
     return undefined;
   }
+
+  private createIdentifier(prefix: string): string {
+    const identifier = `${prefix}${Names.uniqueResourceName(this, { maxLength: 64 - prefix.length })}`;
+    return identifier;
```

Comment:
```suggestion
    return `${prefix}${Names.uniqueResourceName(this, { maxLength: 64 - prefix.length })}`;
```
The function name already clarifies what we are returning.